src/screens/ProfileScreen/elements/RenderItem.js

```diff
@@ -81,24 +81,32 @@ const Item = ({
 
   React.useEffect(() => {
     const validationStatusVote = () => {
-      if (item.reaction_counts) {
-        if (item.latest_reactions.upvotes) {
-          const upvote = item.latest_reactions.upvotes.find((vote) => vote.user_id === selfUserId);
-          if (upvote) {
-            setVoteStatus('upvote');
-            setStatusUpvote(true);
-          }
-        } else if (item.latest_reactions.downvotes) {
-          const downvotes = item.latest_reactions.downvotes.find(
-            (vote) => vote.user_id === selfUserId
-          );
-          if (downvotes) {
-            setVoteStatus('downvote');
-            setStatusDowvote(true);
-          }
-        } else {
-          setVoteStatus('none');
-        }
+      const findUpvote =
+        item &&
+        item.own_reactions &&
+        item.own_reactions.upvotes &&
+        Array.isArray(item.own_reactions.upvotes) &&
+        item.own_reactions.upvotes.find(
+          (reaction) => reaction.user_id === profile.myProfile.user_id
+        );
+      const findDownvote =
+        item &&
+        item.own_reactions &&
+        item.own_reactions.downvotes &&
+        Array.isArray(item.own_reactions.downvotes) &&
+        item.own_reactions.downvotes.find(
+          (reaction) => reaction.user_id === profile.myProfile.user_id
+        );
+      if (findUpvote) {
+        setVoteStatus('upvote');
+        setStatusUpvote(true);
+      }
+      if (findDownvote) {
+        setVoteStatus('downvote');
+        setStatusDowvote(true);
+      }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This logic could potentially set the vote status twice if a user has both upvoted and downvoted an item. While this might not be possible in your application, it's still good practice to handle such edge cases. Consider adding `else if` statements to ensure only one status is set at a time.
```diff
-      if (findUpvote) {
-        setVoteStatus('upvote');
-        setStatusUpvote(true);
-      }
-      if (findDownvote) {
-        setVoteStatus('downvote');
-        setStatusDowvote(true);
-      }

+      if (findUpvote) {
+        setVoteStatus('upvote');
+        setStatusUpvote(true);
+      } else if (findDownvote) {
+        setVoteStatus('downvote');
+        setStatusDowvote(true);
+      }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->